### PR TITLE
[T2A2][W14-A1]Lim Jie

### PR DESCRIPTION
**Pull request by limjie**

### CODE REVIEW

**Comment by limjie** [comment]
Ready for review

File: src/seedu/addressbook/AddressBook.java [hunk 1]
```diff
@@ -315,34 +319,53 @@ private static boolean isValidFilePath(String filePath) {
         if (filePath == null) {
             return false;
         }
-        Path filePathToValidate;
+        return isValid(filePath);
+    }
+
+	private static boolean isValid(String filePath) {
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:325]: isValid is a very vague method name. Are you sure `isValidFilePath` is not adequate?

File: src/seedu/addressbook/AddressBook.java [hunk 2]
```diff
@@ -315,34 +319,53 @@ private static boolean isValidFilePath(String filePath) {
...
-    }
+	}
 
     /**
      * Returns true if the file path has a parent directory that exists.
      */
     private static boolean hasValidParentDirectory(Path filePath) {
         Path parentDirectory = filePath.getParent();
-        return parentDirectory == null || Files.isDirectory(parentDirectory);
+        return isNull(parentDirectory) || isDirectory(parentDirectory);
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:340]: isNull is not necessary actually. There are many different data types that we can check for `null`. would that mean that you need to write many different overloaded `isNull` methods to support this?

File: src/seedu/addressbook/AddressBook.java [hunk 3]
```diff
@@ -315,34 +319,53 @@ private static boolean isValidFilePath(String filePath) {
...
+	private static boolean isValid(String filePath) {
+		Path filePathToValidate;
         try {
             filePathToValidate = Paths.get(filePath);
         } catch (InvalidPathException ipe) {
             return false;
         }
         return hasValidParentDirectory(filePathToValidate) && hasValidFileName(filePathToValidate);
-    }
+	}
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:333]: tab for four space coding violation

File: src/seedu/addressbook/AddressBook.java [hunk 4]
```diff
@@ -1146,12 +1173,16 @@ private static String getUsageInfoForExitCommand() {
     /**
      * Removes sign(p/, d/, etc) from parameter string
      *
-     * @param s  Parameter as a string
-     * @param sign  Parameter sign to be removed
+     * @param fullString  Parameter as a string
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:1176]: good that the method comments are updated

File: src/seedu/addressbook/AddressBook.java [hunk 5]
```diff
@@ -1146,12 +1173,16 @@ private static String getUsageInfoForExitCommand() {
...
+     * @param prefix  Parameter sign to be removed
      * @return  string without the sign
```
Comment (mauris) [inline src/seedu/addressbook/AddressBook.java:1178]: but need to update the `sign` terminology used here.

**Review by mauris — COMMENTED** [review]
it is no good to have so many changes and only one commit. try to aim for one change per commit. it'll pay off.

**Comment by mauris** [comment]
Some comments added. Feel free to add more comment if you have any questions. Once you are done, you may close the PR after reading comments.

**Comment by limjie** [comment]
Woah! Thanks so much! Can't imagine how much effort you put in to see all our changes and comment on them. Really appreciate it!

**Pull request closed without merging**